util/kubeconfig/kubeconfig_test.go

```diff
@@ -161,47 +160,25 @@ func TestNew(t *testing.T) {
 
 	for _, tc := range testCases {
 		caKey, err := certs.NewPrivateKey()
-		if err != nil {
-			t.Fatalf("Failed to generate private key for ca cert, %v", err)
-		}
+		g.Expect(err).NotTo(HaveOccurred())
 
 		caCert, err := getTestCACert(caKey)
-		if err != nil {
-			t.Fatalf("Failed to generate test ca cert, %v", err)
-		}
+		g.Expect(err).NotTo(HaveOccurred())
 
 		actualConfig, actualError := New(tc.cluster, tc.endpoint, caCert, caKey)
 		if tc.expectError {
-			if actualError == nil {
-				t.Fatalf("Expected error but go nil error")
-			} else {
-				continue
-			}
+			g.Expect(actualError).NotTo(HaveOccurred())
+			continue
 		}
 
-		if len(actualConfig.Clusters) != len(tc.expectedConfig.Clusters) {
-			t.Fatalf("Unexpected number of clusters in generated kubeconfig, Want: %d, Got: %d",
-				len(tc.expectedConfig.Clusters), len(actualConfig.Clusters))
-		}
-		if len(actualConfig.Contexts) != len(tc.expectedConfig.Contexts) {
-			t.Fatalf("Unexpected number of contexts in generated kubeconfig, Want: %d, Got: %d",
-				len(tc.expectedConfig.Contexts), len(actualConfig.Contexts))
-		}
-		if _, found := actualConfig.Clusters[tc.cluster]; !found {
-			t.Fatalf("Cluster %q not found in generated kubeconfig", tc.cluster)
-		}
-		if _, found := actualConfig.Contexts[tc.expectedConfig.CurrentContext]; !found {
-			t.Fatalf("Context %q not found in generated kubeconfig", tc.expectedConfig.CurrentContext)
-		}
-		if actualConfig.CurrentContext != tc.expectedConfig.CurrentContext {
-			t.Fatalf("Unexpected value for current config in generated kubeconfig, Want: %q, Got :%q",
-				tc.expectedConfig.CurrentContext, actualConfig.CurrentContext)
-		}
-		if actualConfig.Contexts[tc.expectedConfig.CurrentContext].AuthInfo != tc.expectedConfig.Contexts[tc.expectedConfig.CurrentContext].AuthInfo {
-			t.Fatalf("Unexpected AuthInfo in context for cluter %q in generated kubeconfig, Want: %q, Got: %q",
-				tc.cluster, tc.expectedConfig.Contexts[tc.expectedConfig.CurrentContext].AuthInfo,
-				actualConfig.Contexts[tc.expectedConfig.CurrentContext].AuthInfo)
-		}
+		g.Expect(actualConfig.Clusters).To(HaveLen(len(tc.expectedConfig.Clusters)))
+		g.Expect(actualConfig.Contexts).To(HaveLen(len(tc.expectedConfig.Contexts)))
+
+		g.Expect(actualConfig.Clusters[tc.cluster]).NotTo(BeNil())
+		g.Expect(actualConfig.Contexts[tc.expectedConfig.CurrentContext]).NotTo(BeNil())
+		g.Expect(actualConfig.CurrentContext).To(Equal(tc.expectedConfig.CurrentContext))
+		g.Expect(actualConfig.Contexts).To(Equal(tc.expectedConfig.Contexts))
```

Comment:
Could the order vary of `actualConfig.Contexts`? Might be better to use `ConsistOf` if ordering doesn't matter
```suggestion
		g.Expect(actualConfig.Contexts).To(ConsistOf(tc.expectedConfig.Contexts))
```

Reply:
if use `ConsistOf` the test fail
```
--- FAIL: TestNew (0.29s)
    /Users/cpanato/go/src/github.com/kubernetes-sigs/cluster-api/util/kubeconfig/kubeconfig_test.go:180: 
        Expected
            <map[string]*api.Context | len:1>: {
                "foo-admin@foo": {
                    LocationOfOrigin: "",
                    Cluster: "foo",
                    AuthInfo: "foo-admin",
                    Namespace: "",
                    Extensions: nil,
                },
            }
        to consist of
            <[]interface {} | len:1, cap:1>: [
                {
                    "foo-admin@foo": {
                        LocationOfOrigin: "",
                        Cluster: "foo",
                        AuthInfo: "foo-admin",
                        Namespace: "",
                        Extensions: nil,
                    },
                },
            ]
        the missing elements were
            <[]interface {} | len:1, cap:1>: [
                {
                    "foo-admin@foo": {
                        LocationOfOrigin: "",
                        Cluster: "foo",
                        AuthInfo: "foo-admin",
                        Namespace: "",
                        Extensions: nil,
                    },
                },
            ]
        the extra elements were
            <[]interface {} | len:1, cap:1>: [
                {
                    LocationOfOrigin: "",
                    Cluster: "foo",
                    AuthInfo: "foo-admin",
                    Namespace: "",
                    Extensions: nil,
                },
            ]
FAIL
FAIL	sigs.k8s.io/cluster-api/util/kubeconfig	0.715s
FAIL
Error: Tests failed.
```

Reply:
Ack. Ignore that suggestion 😅 Don't think it works for maps